### ISSUE TITLE
Underscore in hostname

### ISSUE THREAD

**Issue by melihovv** [thephpleague/uri#88]
## Issue summary

Hello. When I try to to create HttpUri object from url, which has underscores in hostname, the `InvalidArgumentException` is thrown with following message `Invalid Hostname, some labels contain invalid characters`. Is there a way to bypass it?

### System informations

| Information | Description |
|--------------|---------|
| League\Uri version |  4.2.0 |
| PHP/HHVM version | 7.0.4  |
| OS Platform | Windows 7 x64  |

## Standalone code, or other way to reproduce the problem
```
$url = 'http://krasnodarskiy_kray.yuginform.ru';
$uri = HttpUri::createFromString($url);
```

**Comment by nyamsprod** [comment]
@melihovv This issue has already been addressed in the previous version of this package. see [Underscore in domain name](https://github.com/thephpleague/url/issues/68).

**TL;DR:** underscore are not allowed in hostname according to [RFC3986](http://tools.ietf.org/html/rfc3986#section-3.2.2).
